### PR TITLE
Add ability to display nested arrays

### PR DESCRIPTION
**Pull request by technetium**
I am targetting this branch, because this can be regarded as either a bugfix or a minor feature.

## Changelog

```markdown

### Changed
- Added `list_array.html.twig` template to include a recursive macro
```

## Subject

When the value of an array contains an array itself, it throws an error
this bugfix/feature will recursively traverse the array and show the complete tree

### CODE REVIEW

File: Resources/views/CRUD/list_array.html.twig [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 This file is part of the Sonata package.
 
-(c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+(c) Thomas Rabaix <thomas.rabaix@sonata-project.org>, Toni Cornelissen <sonata-project@technetium.be>
```
Comment (OskarStark) [inline Resources/views/CRUD/list_array.html.twig]: please remove this line

File: Resources/views/CRUD/list_array_recursion.html.twig [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+This file is part of the Sonata package.
+
+(c) Thomas Rabaix <thomas.rabaix@sonata-project.org>, Toni Cornelissen <toni@iboss.nl>
+
+For the full copyright and license information, please view the LICENSE
+file that was distributed with this source code.
+
+#}
+    {% for key, val in value %}
```
Comment (OskarStark) [inline Resources/views/CRUD/list_array_recursion.html.twig]: indention looks wrong to me

File: Resources/views/CRUD/list_array_recursion.html.twig [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{#
+
+This file is part of the Sonata package.
+
+(c) Thomas Rabaix <thomas.rabaix@sonata-project.org>, Toni Cornelissen <toni@iboss.nl>
```
Comment (OskarStark) [inline Resources/views/CRUD/list_array_recursion.html.twig]: the header is wrong

**Comment by OskarStark** [comment]
Thank you for your contribution! 👍

File: Resources/views/CRUD/list_array.html.twig (hunk 1 above)
Comment (technetium) [inline Resources/views/CRUD/list_array.html.twig]: I thought it was useful having my contact information in the header. I've removed it now.

File: Resources/views/CRUD/list_array_recursion.html.twig (hunk 2 above)
Comment (technetium) [inline Resources/views/CRUD/list_array_recursion.html.twig]: I've lifted most of that code from list_array.html.twig and kept the same indention. I've now de-indented the code so the first line of code starts at the first column.

**Comment by technetium** [comment]
Thanks for your review, I've made the changes I think you've requested.

File: Resources/views/CRUD/list_array.html.twig (hunk 1 above)
Comment (greg0ire) [inline Resources/views/CRUD/list_array.html.twig]: You can use `@author` for that

File: Resources/views/CRUD/list_array.html.twig [hunk 4]
```diff
@@ -12,7 +12,5 @@ file that was distributed with this source code.
 {% extends admin.getTemplate('base_list_field') %}
 
 {% block field %}
-    {% for key, val in value %}
-        [{{ key }} => {{ val }}]
-    {% endfor %}
+    {% include 'SonataAdminBundle:CRUD:list_array_recursion.html.twig' %}
```
Comment (core23) [inline Resources/views/CRUD/list_array.html.twig]: We should use a macro instead of using recursive `include`

**Comment by technetium** [comment]
Thank you for pointing me to macro's, I've replaced the include with a macro.

File: Resources/views/CRUD/list_array.html.twig (hunk 4 above)
Comment (technetium) [inline Resources/views/CRUD/list_array.html.twig]: Replaced the include with a macro.

**Comment by greg0ire** [comment]
You should really read the CONTRIBUTING.md about commit messages. In the meantime, I'm going to try to craft a good one for you.

**Comment by greg0ire** [comment]
The changelog is hard to understand too…

**Comment by greg0ire** [comment]
I edited the changelog, in case anyone is wondering…

**Pull request merged**